src/kbmod/search/stack_search.h

```diff
@@ -50,6 +50,9 @@ class StackSearch {
     // The primary search functions
     void evaluate_single_trajectory(Trajectory& trj);
     Trajectory search_linear_trajectory(short x, short y, float vx, float vy);
+    void prepare_batch_search(std::vector<Trajectory>& search_list, int min_observations);
+    void finish_search();
+    std::vector<Trajectory> search_batch();
```

Comment:
I think you can combine a bunch of duplicate code in `search()` and `search_batch()`.  We keep the common code in the function `search_batch()` by do not have it return the trajectories. We then use two outer functions:

`search_all()`:

1. Call `prepare_batch_search()` (renamed to something like `prepare_search()`)
2. Call `search_batch()`
3. Call `finish_search()`

`search_single_batch()`
1. Call `search_batch()`
2. Return `results.get_batch(0, max_results);`

Reply:
I don't see that many occurrences of `StackSearch::search` across the project (a handful in some notebooks and the main one in `run_search.py`. Are you thinking we should rename all of them to `search_all` in this paradigm?

Reply:
I was thinking that, but perhaps this entire suggestion is over engineering.